src/utils/apis/OfferApi.js

```diff
@@ -74,7 +74,31 @@ export const offersAPI = {
         })
     },
 
+    update: function (offerData) {
+        const offerId = offerData.offerId
```

Comment:
Ogólnie wydaje mi się, że nawet jakbyś takich cyrków z id nie robił, to spring powinien to puścić - pominie nieznane wartości. Jeśli nie, to wystarczy przy OfferRequest wtedy dodać adnotację
```kotlin
@IgnoreJsonProperty(ignoreUnknown = true)
```
A przynajmniej tak mi się wydaje.